src/components/Button/Button.tsx

```diff
@@ -0,0 +1,45 @@
+import { ReactNode } from 'react';
+
+interface ButtonProps {
+  children: ReactNode;
+  className: string;
+  disabled?: boolean;
+  onClick?: () => void;
+  onSubmit?: () => void;
+  type?: 'button' | 'submit';
+}
+
+function Button({
+  children,
+  className,
+  onClick,
+  onSubmit,
+  disabled = false,
+  type = 'button',
+}: ButtonProps) {
+  const classMap: Record<string, string> = {
+    button_large: disabled ? 'button_large_active' : 'button_large_disActive',
+    button_medium: disabled
+      ? 'button_medium_active'
+      : 'button_medium_disActive',
+    button_small: disabled ? 'button_small_active' : 'button_small_disActive',
+    button_large_fill: disabled
+      ? 'button_large_active flex-1'
+      : 'button_large_disActive flex-1',
+  };
+
+  const finalClassName = classMap[className] || className;
```

Comment:
되게 좋은 방법으로 접근하신거같아요 ! 근데 객체로 관리하거나 하나의 변수로 관리하는 방법이 더 깔끔할거 같다고 생각드네요
아래는 따로 생각한 방법이에용 참고만 하시면 좋을거같습니다

```suggestion
const classMap: Record<string, string> = {
  button_large: 'button_large_active button_large_disActive',
  button_medium: 'button_medium_active button_medium_disActive',
  button_small: 'button_small_active button_small_disActive',
  button_large_fill: 'button_large_active flex-1 button_large_disActive flex-1',
};

const finalClassName = disabled 
    ? classMap[className]?.split(' ')[0] 
    : classMap[className]?.split(' ')[1] 
    || className;
```

Reply:
오 되게 좋은방법같애요!! 근데 마지막  button_large_fill 부분이 flex-1 속성을 하나 추가한거라 그부분만 따로 삼항연산자로 넣지나 너무 통일감이 없어보여서 이렇게 했습니다!!